plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend

```diff
@@ -1192,7 +1192,7 @@ class ASTStructureValidator {
 				new DiagnosticMessage(IssueCodes.getMessageForAST_INVALID_CONTINUE,
 					IssueCodes.getDefaultSeverity(IssueCodes.AST_INVALID_CONTINUE), IssueCodes.AST_INVALID_CONTINUE))
 		} else {
-			validateLabel(model, model.label, producer, validLabels)
```

Comment:
As you also rightly said, since `ASTStructureValidator` supposes to validates things that we do not want to do in the grammar, we should only validate that things that do not require resolving proxies here. In the future, consider implement this `validateLabel` in a "normal" `@Check` validator.